src/v/cluster/metadata_cache.cc

```diff
@@ -298,4 +298,24 @@ topic_properties metadata_cache::get_default_properties() const {
     return tp;
 }
 
+ss::future<std::optional<partition_assignment>>
+metadata_cache::get_partition_assignment(
+  const model::ntp& ntp, ss::shard_id shard) const {
+    return _topics_state.invoke_on(shard, [ntp](const topic_table& table) {
+        return table.get_partition_assignment(ntp);
+    });
+}
+
+ss::future<std::optional<std::vector<model::broker_shard>>>
+metadata_cache::get_previous_replica_set(
+  const model::ntp& ntp, ss::shard_id shard) const {
+    return _topics_state.invoke_on(shard, [ntp](const topic_table& table) {
+        return table.get_previous_replica_set(ntp);
+    });
+}
```

Comment:
This doesn't have to accept shard as an argument the state is available on every shard